Update oncall responsibilities (#6757)
fleetdm/fleet: handbook/engineering.md

## Oncall rotation

### Responsibilities

> The oncall engineer is a second-line responder to questions raised by customers and community members. The community contact (Kathy) is responsible for the first response to GitHub issues, pull requests, and Slack messages in the osquery and other public Slacks. Kathy is also responsible for the first response to messages in private customer Slack channels. We respond within 1 hour (during business hours) for interactions and ask the oncall engineer to promptly address any questions sent their way. At times, the oncall engineer may be asked to take over the first response duties when Kathy is unavailable.

Reviewer: ```suggestion
The on-call engineer is a second-line responder to questions raised by customers and community members. The community contact (Kathy) is responsible for the first response to GitHub issues, pull requests, and Slack messages in the osquery and other public Slacks. Kathy is also responsible for the first response to messages in private customer Slack channels. We respond within 1-hour (during business hours) for interactions and ask the on-call engineer to address any questions sent their way promptly. When Kathy is unavailable, the on-call engineer may sometimes be asked to take over the first response duties.
```

> Pull requests may be routed through the oncall engineer for review. For PRs to the Fleet documentation, the content editor (Desmi) will request review from the oncall. For PRs from the community, the community contact will request review from the oncall. In either case, if the oncall has the knowledge and confidence to review, they should go ahead and do so. Otherwise, they should request review from an engineer with the appropriate domain knowledge. It is the oncall's responsibility to ensure that these PRs are moved forward (either by review with feedback or merge).

Reviewer: ```suggestion
Pull requests may be routed through the on-call engineer for review. For PRs to the Fleet documentation, the content editor (Desmi) will request a review from the on-call. For PRs from the community, the community contact will a request review from the on-call. In either case, if the on-call has the knowledge and confidence to review, they should go ahead and do so. Otherwise, they should request a review from an engineer with the appropriate domain knowledge. It is the on-call engineer's responsibility to confirm that these PRs are moved forward (either by review with feedback or merge).
```

> Additionally, the oncall engineer is asked to attend all the customer success meetings during the week. The engineer will receive calendar invitations from Tony for the appropriate meetings. This has a dual purpose of providing more context for how our customers use Fleet, and also the engineer should actively participate and provide input where appropriate (if not sure, please ask your manager or Tony).

Reviewer: ```suggestion
Additionally, the on-call engineer is asked to attend all the customer success meetings during the week. The engineer will receive calendar invitations from Tony for the appropriate meetings. This has a dual purpose of providing more context for how our customers use Fleet. Also, the engineer should actively participate and provide input where appropriate (if not sure, please ask your manager or Tony).
```

### Clearing the plate

> Engineering managers are asked to be aware of the [oncall rotation](https://docs.google.com/document/d/1FNQdu23wc1S9Yo6x5k04uxT2RwT77CIMzLLeEI2U7JA/edit#), and schedule a light workload for engineers while they are oncall. While it varies considerably week by week, the oncall responsibilities can sometimes take up a substantial portion of the engineer's time.

Reviewer: ```suggestion
Engineering managers are asked to be aware of the [on-call rotation](https://docs.google.com/document/d/1FNQdu23wc1S9Yo6x5k04uxT2RwT77CIMzLLeEI2U7JA/edit#) and schedule a light workload for engineers while they are on-call. While it varies week to week considerably, the on-call responsibilities can sometimes take up a substantial portion of the engineer's time.
```

   Click `@oncall`. In the right sidebar, click "Edit Members." Remove the former oncall, and add
   yourself.

> 2. Hand off newer conversations (Slack threads, issues, PRs, etc.). For more recent threads, the former oncall can unsubscribe from the

Reviewer: ```suggestion
2. Handoff newer conversations (Slack threads, issues, PRs, etc.). For more recent threads, the former on-call can unsubscribe from the
```

   thread, and the new oncall should subscribe. The former oncall should explicitly share each of

>    these threads, and the new oncall can select "Get notified about new replies" in the "..." menu.

Reviewer: ```suggestion
   these threads and the new on-call can select "Get notified about new replies" in the "..." menu.
```

   The former oncall can select "Turn off notifications for replies" in that same menu. It can be
   helpful for the former oncall to remain available for any conversations they were deeply involved

>    in, so use your judgment on which threads to hand off. Anything not clearly handed off remains the responsibility of the former oncall.

Reviewer: ```suggestion
   in, so use your judgment on which threads to hand off. Anything not clearly handed off remains the responsibility of the former on-call engineer.
```
